components/bigqueryDS.js

```diff
@@ -0,0 +1,250 @@
+        {
+          trip_id: trip_id,
+          startTime: this.startTime,
+          endTime: this.endTime,
+        }
+      );
+      return [createLogs, updateLogs];
+    });
+
+    const tripLogs = _(await Promise.all(tripPromises))
+      .flatten()
+      .flatten()
+      .map(normalize)
+      .value();
+
+    console.log(`found ${tripLogs.length} ODRD trip logs`);
+    return tripLogs;
+  }
+
+  /*
+   * query update task to find tasks assigned to this vehicle and
+   * then use that result as subquery to find full list of create and
+   * update task requests associated with the specified vehicle.
+   *
+   * Note labels.delivery_vehicle_id is not necessarily set at creation
+   * time (or even on all update requests).
+   */
+  async fetchTaskLogsForVehicle(delivery_vehicle_id) {
+    console.log(
+      "Loading tasks logs for deliveryVehicle id",
+      delivery_vehicle_id
+    );
+    const params = {
+      delivery_vehicle_id,
+      //startTime: this.startTime,
+      //endTime: this.endTime,
+      // XXX ignore start time for field day
+      startTime: new Date(0).toISOString(),
+      endTime: new Date().toISOString(),
+    };
+
+    const createQuery = `
+SELECT
+  *
+FROM
+  \`${this.tablePrefix}create_task\`
+WHERE
+  ${this.timeClause} and labels.task_id IN (
+  SELECT
+    DISTINCT labels.task_id
+  FROM
+    \`${this.tablePrefix}update_task\`
+  WHERE
+    labels.delivery_vehicle_id = @delivery_vehicle_id and ${this.timeClause})
+LIMIT 5000
+`;
+
+    const createData = await this.bq(createQuery, params);
+    const updateQuery = `
+SELECT
+  *
+FROM
+  \`${this.tablePrefix}update_task\`
+WHERE
+  ${this.timeClause} and labels.task_id IN (
+  SELECT
+    DISTINCT labels.task_id
+  FROM
+    \`${this.tablePrefix}update_task\`
+  WHERE
+    labels.delivery_vehicle_id = @delivery_vehicle_id and ${this.timeClause})
+LIMIT 5000
+`;
+
+    const updateData = await this.bq(updateQuery, params);
+
+    const data = createData.concat(updateData);
+    console.log(`found ${data.length} LMFS task logs`);
+    return data.map(normalize);
+  }
+
+  async fetchVehicleLogs(vehicle_id, trip_id) {
+    // TODO: handle case where table doesn't exist (ie in a lmfs project)
+    let whereClause;
+    const params = {
+      startTime: this.startTime,
+      endTime: this.endTime,
+    };
+    if (vehicle_id) {
+      params.vehicle_id = vehicle_id;
+      whereClause = "labels.vehicle_id=@vehicle_id";
+      console.log(`Fetching logs for vehicle ${vehicle_id}`);
+    } else {
+      // XXX why like here?
+      whereClause = `labels.trip_id LIKE @trip_id"`;
+      params.trip_id = `%${trip_id}%`;
+      console.log(`Fetching logs for trip ${trip_id}`);
+    }
+
+    const createData = await this.bq(
+      `select * from \`${this.tablePrefix}create_vehicle\` where ${whereClause} and ${this.timeClause} order by timestamp desc limit 5000`,
```

Comment:
nit: format SQL for readability everywhere as done in L155?
```suggestion
      `
SELECT * FROM \`${this.tablePrefix}create_vehicle\`
WHERE ${whereClause} and ${this.timeClause}
ORDER BY TIMESTAMP DESC LIMIT 5000`,
```

Reply:
Done.